client/src/pages/lessons/edit.tsx

```diff
@@ -476,39 +513,218 @@ const LessonEdit = (props: {
             }}
             variant="outlined"
           />
-          <div className={classes.image}>
-            <TextField
-              data-cy="image"
-              label="Image"
-              placeholder="Link to image url"
-              multiline
-              rowsMax={4}
-              fullWidth
-              InputLabelProps={{
-                shrink: true,
-              }}
-              value={lessonUnderEdit.lesson?.image || ""}
-              onChange={(e) => {
-                setLesson(
-                  {
-                    ...(lessonUnderEdit.lesson || newLesson),
-                    image: e.target.value || "",
-                  },
-                  true
-                );
-              }}
-              variant="outlined"
-            />
-            <img
-              className={classes.thumbnail}
-              data-cy="image-thumbnail"
-              src={lessonUnderEdit.lesson?.image}
-              style={{ height: 50 }}
-              onClick={() => {
-                window.open(lessonUnderEdit.lesson?.image, "_blank");
+          <FormControl
+            style={{ width: 800 }}
+            className={classes.selectForm}
+            variant="outlined"
+          >
+            <InputLabel shrink id="media-label">
+              Media Type
+            </InputLabel>
+            <Select
+              labelId="media-label"
+              data-cy="media-type"
+              value={lessonUnderEdit.lesson.media.type ?? MediaType.NONE}
```

Comment:
we should almost always prefer the `||` check over the null-coalescing operator `??`

The difference is that `??` doesn't care about falsyness (e.g. an empty string).

Let's go back and replace all new `??` use with `||` unless there's a specific reason for `??` (and note why in the comments)